### ISSUE TITLE
Background slideshow gone on login screen

### ISSUE THREAD

**Issue by mikehaertl** [linuxmint/mint-mdm-themes#24]
On my 18.2 Sonya installation the slideshow on the login screen suddenly is gone. Doing some googling I found, that it's probably configured in `/usr/share/mdm/html-themes/Mint-X/slideshow.conf`. The content for me is something like:

```
...
file:///usr/share/backgrounds/linuxmint-serena/ajilden_blossom.jpg
file:///usr/share/backgrounds/linuxmint-serena/anieto2k_hot_balloon_trip.jpg
file:///usr/share/backgrounds/linuxmint-serena/arestifo_hill.jpg
file:///usr/share/backgrounds/linuxmint-serena/capelli_ears_of_corn.jpg
...
```

None of those exists on my system. The content of my `/usr/share/backrounds/` directory is:

```
drwxr-xr-x   2 root root 4,0K Jul  4 09:44 linuxmint
drwxr-xr-x   2 root root 4,0K Jun 28  2016 linuxmint-sarah
drwxr-xr-x   2 root root 4,0K Jul  4 09:44 linuxmint-sonya
```

**Comment by DennisLfromGA** [comment]
@mikehaertl,

Same thing happened to me and probably many others.

There is a config file that contains a path to the background images for each release here -

- `/usr/share/mdm/html-themes/Mint-X/slideshow.conf`

And there's a directory with the images for sonya here -

- `/usr/share/backgrounds/linuxmint-sonya/`

I grabbed the files and formatted them with `sed` -

- `ls -1 /usr/share/backgrounds/linuxmint-sonya/* | grep -v Credits | sed 's.^.file://.' > /tmp/sonya.bg`

I first made a backup of the `slideshow.conf` file and then edited it as root and appended the `/tmp/sonya.bg` file.

I also commented out all of the backgrounds from the earlier releases.

Once I logged out again, I got the Sonya slideshow. :+1: 

When they fix this bug it may break or change this workaround but it's good for now at least.

*Hope this helps,*
-**DennisL**

**Comment by mikehaertl** [comment]
@DennisLfromGA That's what I did, too. Another fix is to install the background packages for older releases, especially mint-backgrounds-serena.

Still think, that this should be fixed, as the background slideshow is very likely completely broken on fresh installments.

**Comment by camypaj** [comment]
I've created a pull request for this, so let's see if @clefebvre or someone from the dev team picks it up.
Cheers

**Comment by camypaj** [comment]
Hi @mikehaertl , my pull request should fix the issue, but I'm not sure when and how this gets published further. Feel free to check/test it out if you can/want, but I believe as this gets published, we can close the issue.
Regards

**Comment by clefebvre** [comment]
Hi sorry about this.

I'm not sure why you don't have mint-backgrounds-serena, it's installed by default in 18.1 and 18.2... the only way I guess you wouldn't have it was if you upgraded from 18.0 to 18.2?

Fresh installs are ok, they have that package, and they also use lightdm by default. Still, thanks for catching our attention on this, this was fixed today for the upcoming 18.3.

**Comment by mikehaertl** [comment]
@clefebvre I've followed the usual upgrade path and always upgraded to the latest release whenever it was ready over the last couple of years. Still I was hit by this problem here. So somewhere the serena themes must have been uninstalled automatically. I did not uninstall them.

**Comment by camypaj** [comment]
Hi @clefebvre , for me it was the same - I've mostly upgraded as well, and I didn't remove any packages, at least not manually or on purpose.
Actually, I was thinking about it yesterday morning - maybe the best idea would be to list all wallpapers from all supported releases in the config, so that everybody using a supported release would get at least some wallpapers? Another option would be to have a script that generates this config, on every X event? For example, reboot? Unless it's an overkill/overengineering :)
Third one would be to have one default/fallback wallpaper, which should always be there?

Regards

**Comment by xenopeek** [comment]
> Fresh installs are ok, they have that package, and they also use lightdm by default. Still, thanks for catching our attention on this, this was fixed today for the upcoming 18.3.

From that comment concluding that this issue was fixed so closing it here.
